src/subapps/projects/components/ResourcesPane.tsx

```diff
@@ -34,44 +39,62 @@ const ResourcesPane: React.FC<{}> = ({ children }) => {
     };
   }, []);
 
-  const onClickLinkCode = (event: any) => {};
+  // TODO: link notes
+  const onClickAddNotes = () => {};
 
-  const onClickAddNotes = (event: any) => {};
+  const onClickSave = (data: any) => {
+    linkCode(data);
+    setShowCodeForm(false);
+  };
 
   return (
-    <div
-      ref={paneRef}
-      className="resources-pane"
-      style={{ width: `${paneWidth}px` }}
-    >
-      <Collapse>
-        <Panel
-          header={
-            <div className="resources-pane__header">
-              <span className="resources-pane__pane-name">Resources</span>
-              <ActionButton
-                title="Link code"
-                onClick={() => onClickLinkCode}
-                icon="Add"
-              />
-              <ActionButton
-                title="Add notes"
-                onClick={() => onClickAddNotes}
-                icon="Add"
-              />
-              <ActionButton
-                title="Link or add data"
-                onClick={() => {}}
-                icon="Add"
-              />
-            </div>
-          }
-          key="1"
-        >
-          <div className="resources-pane__content">{children}</div>
-        </Panel>
-      </Collapse>
-    </div>
+    <>
+      <div
+        ref={paneRef}
+        className="resources-pane"
+        style={{ width: `${paneWidth}px` }}
+      >
+        <Collapse>
+          <Panel
+            header={
+              <div className="resources-pane__header">
+                <span className="resources-pane__pane-name">Resources</span>
+                <ActionButton
+                  title="Link code"
+                  onClick={() => setShowCodeForm(true)}
```

Comment:
it's better to reference a variable here, and not write an arrow function, for increased performance.

Reply:
I think not having an arrow function will befit when you have set of elements and we are adding arrow functions to them on a loop. For an element like the above, it looks fine to me.